utils/ansible.py

```diff
@@ -0,0 +1,281 @@
+    rmtree(local_git_repo)
+
+
+def get_values_from_conf(provider, script_type):
+    if script_type == 'providers':
+        return {
+            'name': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('name', []),
+            'state': 'present',
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+            'provider_api_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', []),
+            'provider_api_auth_token': get_crud('openshift-3').credentials['token'].token,
+            'hawkular_hostname': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('hostname', [])
+        }
+    elif script_type == 'users':
+        return {
+            'fullname': 'MIQUser',
+            'name': 'MIQU',
+            'password': 'smartvm',
+            'state': 'present',
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+    elif script_type == 'custom_attributes':
+        return {
+            'entity_type': 'provider',
+            'entity_name': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('name', []),
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+
+
+# TODO Avoid reading files every time
+def read_yml(script, value):
+    with open(yml_path + script + yml, 'r') as f:
+            doc = load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def get_yml_value(script, value):
+    with open(basic_yml_path + script + yml, 'r') as f:
+            doc = load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def setup_basic_script(provider, script_type):
+    script_path = path.join(basic_yml_path, script_type + basic_script)
+    script_path_bk = path.join(basic_yml_path, script_type + basic_script_bk)
+    copyfile(script_path, script_path_bk)
+    with open(script_path, 'rw') as f:
+        doc = load(f)
+        values_dict = get_values_from_conf(provider, script_type)
+    for key in values_dict:
+        if script_type == 'providers':
+            doc[0]['tasks'][0]['manageiq_provider'][key] = values_dict[key]
+        elif script_type == 'users':
+            doc[0]['tasks'][0]['manageiq_user'][key] = values_dict[key]
+        elif script_type == 'custom_attributes':
+            doc[0]['tasks'][0]['manageiq_custom_attributes'][key] = values_dict[key]
+        with open(basic_yml_path + script_type + basic_script, 'w') as f:
+            f.write(dump(doc))
+
+
+def open_yml(script, script_type):
+    copyfile(basic_yml_path + "/" + script_type + basic_script,
```

Comment:
**REQUIRED:** please use `os.path.join` or `py.path.local` (as used in `utils.path`) for all path joining to avoid potential problems, for supporting later maintenance i would advise for `py.path.local` as it would ensure uniform usage